lib/services/cloud-project-service.ts

```diff
@@ -87,26 +86,40 @@ export class CloudProjectService extends CloudService implements ICloudProjectSe
 
 		const tasksResults: IDictionary<IServerResult> = {};
 		this.$logger.info(`${EOL}### Build machines cleanup${EOL}`);
-		for (let i = 0; i < cleanupResponse.buildMachineResponse.length; i++) {
-			const taskId = `Cloud Cleanup Task #${i + 1}`;
-			const task = cleanupResponse.buildMachineResponse[i];
-			try {
-				await this.waitForServerOperationToFinish(taskId, task);
-				const taskResult = await this.getObjectFromS3File<IServerResult>(task.resultUrl);
-				const output = this.$nsCloudOutputFilter.filter(await this.getContentOfS3File(task.outputUrl));
-				if (this.hasContent(output)) {
-					this.$logger.info(output);
+		const childTasks: Promise<IChildTaskExecutionResult>[] = [];
+		for (let task of cleanupResponse.buildMachineResponse) {
+			const executeChildTask = async (): Promise<IChildTaskExecutionResult> => {
+				const childCloudOperationId = task.cloudOperationId;
+				let output = `Child cloud operation id: ${childCloudOperationId}`;
+				try {
+					const taskResult = await this.waitForServerOperationToFinish(childCloudOperationId, task, { silent: true });
+					tasksResults[childCloudOperationId] = taskResult;
+					output += await this.getCollectedLogs(task);
+					return { output };
+				} catch (err) {
+					if (this.getResult(childCloudOperationId)) {
+						tasksResults[childCloudOperationId] = this.getResult(childCloudOperationId);
```

Comment:
Extract this.getResult(childCloudOperationId) to a const in order to remove repetition

Reply:
Done.